iac-extensions/kubernetes/src/test/java/org/sonar/iac/kubernetes/checks/DuplicatedEnvironmentVariablesCheckTest.java

```diff
@@ -43,30 +46,75 @@ static Stream<String> sensitiveKinds() {
+      "DuplicatedEnvironmentVariables/CrossFileTests/NonCompliant/RepetitionWithKindTemplate/my-secret.yaml");
+  }
+
   @Test
   void shouldVerifyHelmPod() {
     // secondary location is the same line as primary location
-    var expectedSecondary1 = new SecondaryLocation(range(19, 0, 19, 35), SECONDARY_MESSAGE);
+    var expectedSecondary1 = new SecondaryLocation(range(19, 0, 19, 35), SECONDARY_MESSAGE_VARIABLE);
     var expectedSecondary2 = new SecondaryLocation(range(12, 2, 18, 0), "This value is used in a noncompliant part of a template",
-      "DuplicatedEnvironmentVariables/DuplicatedEnvsChart/values.yaml");
+      "DuplicatedEnvironmentVariables/SingleFileTests/DuplicatedEnvsChart/values.yaml");
     // secondary location is the same line as primary location
-    var expectedSecondary3 = new SecondaryLocation(range(28, 0, 28, 55), SECONDARY_MESSAGE);
-    var expectedSecondary4 = new SecondaryLocation(range(46, 0, 46, 20), SECONDARY_MESSAGE);
-    var expectedSecondary5 = new SecondaryLocation(range(48, 0, 48, 20), SECONDARY_MESSAGE);
+    var expectedSecondary3 = new SecondaryLocation(range(28, 0, 28, 55), SECONDARY_MESSAGE_VARIABLE);
+    var expectedSecondary4 = new SecondaryLocation(range(46, 0, 46, 20), SECONDARY_MESSAGE_VARIABLE);
+    var expectedSecondary5 = new SecondaryLocation(range(48, 0, 48, 20), SECONDARY_MESSAGE_VARIABLE);
     var expectedIssues = List.of(
-      issue(19, 19, 19, 24, PRIMARY_MESSAGE, expectedSecondary1),
-      issue(28, 10, 28, 31, PRIMARY_MESSAGE, expectedSecondary2, expectedSecondary3),
-      issue(50, 16, 50, 20, PRIMARY_MESSAGE, expectedSecondary4, expectedSecondary5));
-    KubernetesVerifier.verify("DuplicatedEnvironmentVariables/DuplicatedEnvsChart/templates/duplicated-env-pod.yaml", check, expectedIssues);
+      issue(19, 19, 19, 24, PRIMARY_MESSAGE_VARIABLE, expectedSecondary1),
+      issue(28, 10, 28, 31, PRIMARY_MESSAGE_VARIABLE, expectedSecondary2, expectedSecondary3),
+      issue(50, 16, 50, 20, PRIMARY_MESSAGE_VARIABLE, expectedSecondary4, expectedSecondary5));
+    KubernetesVerifier.verify("DuplicatedEnvironmentVariables/SingleFileTests/DuplicatedEnvsChart/templates/duplicated-env-pod.yaml", check, expectedIssues);
+  }
+
+  @ParameterizedTest
+  @CsvSource(delimiter = ';', value = {
+    "RepetitionAcrossAllSources;repetition_across_all_sources.yaml;my-config-map.yaml,my-secret.yaml",
+    "RepetitionAcrossTwoConfigMap;repetition_across_two_config_map.yaml;my-config-map-1.yaml,my-config-map-2.yaml",
+    "RepetitionAcrossTwoConfigMapExplicitNamespace;repetition_across_two_config_map_explicit_namespace.yaml;my-config-map-1.yaml,my-config-map-2.yaml",
+    "RepetitionAcrossTwoSecret;repetition_across_two_secret.yaml;my-secret-1.yaml,my-secret-2.yaml",
+    "RepetitionInsideConfigMapAfter;repetition_inside_config_map_after.yaml;my-config-map.yaml",
+    "RepetitionInsideConfigMapBefore;repetition_inside_config_map_before.yaml;my-config-map.yaml",
```

Comment:
Is it not better to use `Arguments.arguments()` instead of CSV? It is hard to read the above CSV.